spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskExecutionThinController.java

```diff
@@ -44,15 +48,29 @@ public class TaskExecutionThinController {
 	private final AggregateTaskExplorer explorer;
 	private final TaskExecutionThinResourceAssembler resourceAssembler;
 
-	public TaskExecutionThinController(AggregateTaskExplorer explorer) {
+	private final TaskJobService taskJobService;
+
+	public TaskExecutionThinController(AggregateTaskExplorer explorer, TaskJobService taskJobService) {
 		this.explorer = explorer;
+		this.taskJobService = taskJobService;
 		this.resourceAssembler = new TaskExecutionThinResourceAssembler();
 	}
 
 	@GetMapping(produces = "application/json")
 	@ResponseStatus(HttpStatus.OK)
 	public PagedModel<TaskExecutionThinResource> listTasks(Pageable pageable, PagedResourcesAssembler<AggregateTaskExecution> pagedAssembler) {
-		return pagedAssembler.toModel(explorer.findAll(pageable, true), resourceAssembler);
+		Page<AggregateTaskExecution> page = explorer.findAll(pageable, true);
+		taskJobService.populateComposeTaskRunnerStatus(page.getContent());
```

Comment:
Have we determined the cost of running the retrieval of the CTR status for the results set?

Reply:
I compared before and after timings from `org.springframework.cloud.dataflow.integration.test.tasks.TaskExecutionQueryIT` 
I added index in parent_execution_id and now the difference is hardly noticeable.